### DISCUSSION TITLE
Import on a different machine to my docker container

### DISCUSSION THREAD

**Discussion by davidcreager** [mediagis/nominatim-docker#306] [Q&A] [answered]
Hi,
Apologies for the probably stupid question,.

I have run the import on a stand alone machine and now want to move the necessary files to the docker container.   

I assume I need to move  the PROJECT_DIR but I can't see how to tell docker where it is!!

**Comment by leonardehrenfried** [comment]
I think I don't understand what you're trying to do.

If you have completed the import then the files are already in the container, namely in the postgres database.

**Comment by davidcreager** [comment]
Argh,   Sorry I was unclear.     The machine that I use for docker is not well specced enough to undertake the import, so I ran it on completely different hardware.   I now want to have nominatim running on my docker machine in a docker container but using the import I created on the different hardware

**Comment by leonardehrenfried** [comment]
For that I would use the docker import/export functionality: https://bobcares.com/blog/docker-export-and-import-containers/

**Reply by davidcreager** [reply]
Ah  Ok thanks.  I guess I will attempt to get docker running on my faster machine

I was hoping to make use of 'Moving the database to another machine' from the nominatim documentation
https://nominatim.org/release-docs/latest/admin/Advanced-Installations/#moving-the-database-to-another-machine

**Comment by leonardehrenfried** [comment]
Well, the pgdump way would still work. You need to open the postgres port and point pgdump to use the postgres instance inside the container.

I have never tried it with nominatim though.

**Reply by davidcreager** [reply]
Ok thanks for all your help

**Comment by davidcreager** [comment, marked as answer]
Ok...So I finally achieved what I wanted. In case it helps anyone else. (I am sure this is not optimum but hey ho it worked at least for me)
My Objective was to create Nominatim on a non-docker pc and then move the database to a docker container running on a separate pc.
(I did not want my docker PC to undertake the load of the first import).

Created the docker container using the smallest (Andorra) map (2mb). This took a minute or so. I set the container with the persistence config as per standard instructions.
Ran the install and import (of Great Britain) on my separate non-docker pc (took about 6 hours).
Copied /var/lib/postgresql/12/main from the non-docker pc to a location that I could be accessed by a mount in a docker container.
Created a mount in the nominatim docker for the location where the copied /var/lib/postgresql/12/main was
Created a terminal to the running docker container and changed the ownership and group of the copied /var/lib/postgresql/12/main to be postgres.

**Reply by shinhanshiu** [reply]
Thanks for the solution. I tried to repeat what you have outlined but was not successful. I tried this with Andorra as the recipient and Equatorial Guinea as the donor. 

The main issue is that I am not sure how you can "Created a mount in the nominatim docker for the location":
- I tried this by passing -v parameter in the docker run call but the database was overwritten completely.
- I then tried this by using "ro" paramters for bind mount, but this led to "fatal: creating new database failed".
- I then copy the main folder from the donor into the running recipient container's main folder, then change the ownership but this lead to php fatal error.
- I also try to stop the container, copy the folder in, change the ownership, but the container cannot be restarted with docker start.

I wonder exactly how you did this. Thanks.

**Reply by davidcreager** [reply]
Aha.   Docker can be very fiddly with this stuff.   It's been a while since I did this, but looking at my container, it seems I created a mount with a label not used by nominatim (in my case, I chose 'cpdata') , then I exec'd into the running container and copied from this label to the volume used by nominatim for its database. 

I am not sure if 'exec' is the correct term,  I use Unraid and that allows you to start a container and open a console (terminal shell)

Hope this helps
